### PR TITLE
Channel/privet/open create

### PR DESCRIPTION
**Pull request by wael85**
# Description

I create endpoint that can send bake common channel between tow user

Fixes # (issue)

# How to test?

Please provide a short summary how your changes can be tested?

# Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] This PR is ready to be merged and not breaking any other functionality

### CODE REVIEW

File: src/server/api/controllers/private-channel.controller.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+const knex = require('../../config/db');
+// const Error = require('../lib/utils/http-error');
+// const moment = require('moment-timezone');
+
+const checkPrivateChannel = async (userId1, userId2) => {
```
Comment (benna100) [inline src/server/api/controllers/private-channel.controller.js:5]: This needs to support multiple users, what if there were 5 users that i wanted to check fx

File: src/server/api/controllers/private-channel.controller.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  const result = await knex('channel_members as a')
+    .select('a.fk_channel_id', 'a.fk_user_id')
+    .join('channel_members as b', 'b.fk_channel_id', 'a.fk_channel_id')
+    .groupBy('a.id')
+    .havingIn('a.fk_user_id', [userId1, userId2]);
```
Comment (benna100) [inline src/server/api/controllers/private-channel.controller.js:10]: Awesome, well done

File: src/server/api/controllers/private-channel.controller.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+const knex = require('../../config/db');
+// const Error = require('../lib/utils/http-error');
+// const moment = require('moment-timezone');
+
+const checkPrivateChannel = async (userId1, userId2) => {
+  const result = await knex('channel_members as a')
```
Comment (benna100) [inline src/server/api/controllers/private-channel.controller.js:6]: result vague variable name, please change

File: src/server/api/routes/private-channel.router.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+ *            classId:
+ *              type: string
+ *    responses:
+ *      201:
+ *        description: private channel created
+ *      5XX:
+ *        description: Unexpected error.
+ */
+router.get('/', (req, res) => {
+  const { user1, user2 } = req.query;
```
Comment (benna100) [inline src/server/api/routes/private-channel.router.js:44]: support infinite number of users!

**Review by benna100 — CHANGES_REQUESTED** [review]
Damn you have done a lot since i gave you feedback, well done!

There are no private channels, just channels. This functionality should lay in the channel controller. 

Apart from that this is a super good PR wael, there are some changes, but its tight! Only relevant files, linked issue, great job man!!

File: src/server/api/controllers/private-channel.controller.js (hunk 2 above)
Comment (panivita) [reply src/server/api/controllers/private-channel.controller.js:10]: Hi Wael. I check that in database. It is not return channel, where 2 or more users. It is return the same information about channels. If you want select channel, where was 2 or more users, this work in database
select fk_channel_id from channel_members
where fk_user_id in (1,3)
group by fk_channel_id
having count(*) = 2;
you can change user_id and add more and change count('*) 
In knex you should think how add userIds and count
What do you think?

File: src/server/api/routes/api-router.js [hunk 5]
```diff
@@ -37,10 +40,13 @@ const swaggerDocument = swaggerJsDoc(swaggerOptions);
 router.use('/documentation', swaggerUi.serve, swaggerUi.setup(swaggerDocument));
 
 // Application routes
+router.use('/private-channel/', privateChannelRouter);
 router.use('/modules', modulesRouter);
 router.use('/users', usersRouter);
 router.use('/channels', channelsRouter);
 router.use('/messages', messagesRouter);
+router.use('/channel-members', channelMembersController);
```
Comment (benna100) [inline src/server/api/routes/api-router.js:48]: lint error here

Comment (tpetrina) [inline src/server/api/routes/api-router.js:48]: I guess these two lines are done by mistake? They are duplicate!

**Review by benna100 — APPROVED** [review]
Looks good, you are still adding quite some things  that is not related to your branch, but let's get this into develop

**Comment by benna100** [comment]
Ahh there are merge conflicts. I'm sure it's because of the changes you add that are not relevant to your branch

File: src/server/api/routes/api-router.js [hunk 6]
```diff
@@ -14,6 +14,9 @@ const messagesRouter = require('./messages-router');
 const channelMembersRouter = require('./channel-members.router');
 const notificationsRouter = require('./notifications-router');
 
+// private channel router imports
+const privateChannelRouter = require('./private-channel.router');
```
Comment (benna100) [inline src/server/api/routes/api-router.js:18]: How is this not throwing an error? There is no private channel router right!? You have to check these things wael!

File: src/server/api/routes/channel-members.router.js [hunk 7]
```diff
@@ -110,10 +110,51 @@ router.post('/', (req, res) => {
...
+ *        content:
+ *          application/json:
+ *             schema:
+ *               type : array
+ *               items:
+ *                  type : integers
+ *      5XX:
+ *        description: Unexpected error.
+ */
+router.get('/?users[]', (req, res) => {
```
Comment (benna100) [inline src/server/api/routes/channel-members.router.js:146]:
```suggestion
router.get('/is-part-of-channel', (req, res) => {
```

**Pull request merged**